### ISSUE TITLE
Switch to window when releasing alt-tab without needing to click window

### ISSUE THREAD

**Issue by ericjung** [lwouis/alt-tab-macos#465]
Perhaps this is already possible, but I did not find it.

Currently, in order to switch to a different window, the user presses alt-tab to display a thumbnail list of windows. Subsequent alt-tab keypresses cycles through highlighting the thumbnails, but releasing alt-tab does not bring a window to the foreground ass it does on Windows. The user must instead mouse-click one of the windows to foreground it. 

**Describe the solution you'd like**
Releasing alt-tab would activate the selected window and bring it to the foreground.

**Describe alternatives you've considered**
I've looked through the UI to see if there is an option for this behavior. I briefly looked at open github issues to see if this is already requested (although I did not read through all 42 of them, to be honest).

I apologize if this has been suggested before; I'm sure it has -- I just don't see anything about it.

<details>
<summary><strong>Debug profile</strong></summary>
<p>

* App version: 4.9.1
* App preferences: 
  * MSAppCenter310AppCenterUserDefaultsMigratedKey: 1
  * MSAppCenter310CrashesUserDefaultsMigratedKey: 1
  * MSAppCenterInstallId: C6905461-6A44-4066-8F55-F162F44C8CC5
  * MSAppCenterPastDevices: {length = 1133, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 0000037b }
  * MSAppCenterSessionIdHistory: {length = 452, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 0000014e }
  * MSAppCenterUserIdHistory: {length = 447, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 00000149 }
  * NSWindow Frame SUUpdateAlert: 650 494 620 413 0 0 1920 1057 
  * NSWindow Frame com.sindresorhus.Preferences.FrameAutosaveName: 790 445 340 267 0 0 1920 1057 
  * SUHasLaunchedBefore: 1
  * SULastCheckTime: 2020-07-22 20:18:07 +0000
  * SUUpdateGroupIdentifier: 3997019679
  * SUUpdateRelaunchingMarker: 0
  * appsToShow: 0
  * cancelShortcut: ⎋
  * closeWindowShortcut: C
  * hideShowAppShortcut: H
  * holdShortcut: ⌥
  * preferencesVersion: 4.9.1
  * screensToShow: 0
  * spacesToShow: 0
  * theme: 1
  * updatePolicy: 1
* Applications: 50
* Windows: 17
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: true, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: true, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
  * {isFullscreen: false, isMinimized: false, isHidden: false, isTabbed: false, isOnAllSpaces: false, spaceId: 1, spaceIndex: 1}
* OS version: Version 10.15.5 (Build 19F101)
* OS architecture: x86_64
* Locale: en_US (current)
* Spaces: 2
* Dark mode: Dark
* "Displays have separate Spaces": checked
* Hardware model: MacBookPro16,1
* Screens: 1
  * {x: 0.0, y: 0.0, width: 1920.0, height: 1080.0}
* CPU model: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
* Memory size: 34.36 GB
* Active CPU count: 16
* Current CPU frequency: 2.3 Ghz
* Resource utilization: 
  * CPU: 0.5%
  * Memory: 93M
  * Threads count: 12

</p>
</details>

**Comment by lwouis** [comment]
Hi @ericjung! Thanks for sharing this issue!

What you describe seems to be a bug. AltTab uses the same keyboard scheme as Windows: hold alt > tab, tab, tab > release alt to focus window.

To help facilitate the investigation, could you please click the "send feedback" button on the menubar icon, and keep the "send debug profile" checkbox checked? That will open another ticket here with tech info like which version of macOS and AltTab you use.

Thanks!

**Comment by ericjung** [comment]
Hi @lwouis,

I tried sending feedback as you describe, but it does not work. Clicking the `send` button does nothing. I've tried this on 2 different macbooks pros. Both have the same problem I describe in the opening of this ticket. Both use version 4.9.1. One is using macOS 10.15.5 and the other 10.14.6. Both have 32 GB RAM.

Here's a screenshot:

![image](https://user-images.githubusercontent.com/5875610/88316743-83d38800-ccd5-11ea-9aa4-381a654b0b1a.png)

The `send feedback` window behaves strangely for input: there is no cursor in the text area where I've typed (so I cannot see where the next character will emit). Hovering over `cancel` or `send` buttons do not change the mouse cursor as I'd expect. And clicking `send` or `cancel` just does nothing at all. I tried entering an email address. Same problem.

What else can I do to get you the information you need?

**Comment by SoPat712** [comment]
You can try sending him as much information as possible. Try giving him the model of your Mac, MacOS version, Alt-tab Version, how many CPU cores you have, GB of ram that you have, storage size of your Mac, free space available, etc.

**Comment by ericjung** [comment]
> free space available

How can this possibly be relevant?

**Comment by lwouis** [comment]
That's a pretty crazy situation. Here are the problems I gather from your reports:

* Releasing `alt` doesn't focus the selected window
* In the `Send feedback` form, after focusing the big text area, there is no visible blinking cursor, as is normally the case for text input fields
* In the `Send feedback` form, when hovering the `Cancel` and `Send` buttons, the cursor stays as if editing text (i.e. a vertical bar, instead of a pointer)

Also noteworthy:

* These 3 issues are reproduced on 2 different systems
* None of these issues involve AltTab crashing. The app is running, its icon visible and clickable in the menubar

On my local machine, none of these issues happen. It also seems to be working fine on most people system. Thus I'm interested to know your:

* macOS version
* AltTab version
* Preferences (you can get these by typing `defaults read com.lwouis.alt-tab-macos` in Terminal.app)

Also, could you please check in Console.app, if there are logs related to AltTab? You just need to open the app, type "AltTab" in the search bar on the top right, then copy-paste lines you get here.

Of course, if you have any other intuition on what could be special about your systems would help (e.g. special hardware like a keyboard, apps to remap keys, etc)

**Comment by ericjung** [comment]
Thanks for the detailed reply. I agree this is rather crazy, especially since it's reproducible on two completely independent machines. I do have a 3rd macbook on which I can try AltTab if you want.

> In the Send feedback form, after focusing the big text area, there is no visible blinking cursor, as is normally the case for text input fields
In the Send feedback form, when hovering the Cancel and Send buttons, the cursor stays as if editing text (i.e. a vertical bar, instead of a pointer)

After restarting the machine, I was able to use the `send feeback` button (see issue #470) but yes, initially that is what I saw.

Restart did not fix `Releasing alt doesn't focus the selected window` 

> Also noteworthy:
These 3 issues are reproduced on 2 different systems
None of these issues involve AltTab crashing. The app is running, its icon visible and clickable in the menubar

Correct

> Thus I'm interested to know your:
macOS version
AltTab version

I as I wrote earlier, both use version 4.9.1. One is using macOS 10.15.5 and the other 10.14.6. Both have 32 GB RAM and i9 CPUs. These are beefy machines and the problem is not related to performance since it happens with most all applications closed (except alt-tab).

> Preferences (you can get these by typing defaults read com.lwouis.alt-tab-macos in Terminal.app)

One machine 1 (macOS 10.15.5):

```
defaults read com.lwouis.alt-tab-macos
{
    MSAppCenter310AppCenterUserDefaultsMigratedKey = 1;
    MSAppCenter310CrashesUserDefaultsMigratedKey = 1;
    MSAppCenterInstallId = "C6905461-6A44-4066-8F55-F162F44C8CC5";
    MSAppCenterPastDevices = {length = 1133, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 0000037b };
    MSAppCenterSessionIdHistory = {length = 452, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 0000014e };
    MSAppCenterUserIdHistory = {length = 447, bytes = 0x62706c69 73743030 d4010203 04050607 ... 00000000 00000149 };
    "NSWindow Frame SUUpdateAlert" = "650 494 620 413 0 0 1920 1057 ";
    "NSWindow Frame com.sindresorhus.Preferences.FrameAutosaveName" = "790 445 340 267 0 0 1920 1057 ";
    SUHasLaunchedBefore = 1;
    SULastCheckTime = "2020-07-22 20:18:07 +0000";
    SUUpdateGroupIdentifier = 3997019679;
    SUUpdateRelaunchingMarker = 0;
    appsToShow = 0;
    cancelShortcut = "\\U238b";
    closeWindowShortcut = C;
    hideShowAppShortcut = H;
    holdShortcut = "\\U2325";
    preferencesVersion = "4.9.1";
    screensToShow = 0;
    spacesToShow = 0;
    theme = 1;
    updatePolicy = 1;
}
```

> could you please check in Console.app, if there are logs related to AltTab? You just need to open the app, type "AltTab" in the search bar on the top right, then copy-paste lines you get here.

On machine 1: 

There are 152 messages after leaving Console open for a bit. I can't copy/paste them but here's an excerptof some messages  (ignoring the numerous duplicates):

> #Error Unexpected key code 0 for the FlagsChanged event
-[TCCDAccessIdentity staticCode]: static code for: identifier com.lwouis.alt-tab-macos, type: 0: 0x7fec764175e0 at /Applications/AltTab.app
Acquiring assertion targeting executable<AltTab(501)> from originator [daemon<com.apple.coreservices.launchservicesd>:144] with description <RBSAssertionDescriptor; frontmost:84649; ID: 339-144-8294; target: 84649> attributes = {
    <RBSDomainAttribute: 0x7fb6ffa24ae0; domain: com.apple.launchservicesd; name: RoleUserInteractiveFocal; sourceEnvironment: 0x0>;
}
Assertion 339-144-8294 (target:executable<AltTab(501)>) will be created as active
Finished acquiring assertion 339-144-8294 (target:executable<AltTab(501)>)
[executable<AltTab(501)>:84649] Ignoring jetsam update because this process is not memory-managed
[executable<AltTab(501)>:84649] Set darwin role to: UserInteractiveFocal
[executable<AltTab(501)>:84649] Ignoring GPU update because this process is not GPU managed
Invalidating assertion 339-144-8294 (target:executable<AltTab(501)>) from originator 144
[executable<AltTab(501)>:84649] Set darwin role to: UserInteractive
Acquiring assertion targeting executable<AltTab(501)> from originator [daemon<com.apple.coreservices.launchservicesd>:144] with description <RBSAssertionDescriptor; frontmost:84649; ID: 339-144-8297; target: 84649> attributes = {
    <RBSDomainAttribute: 0x7fb6ff81a6d0; domain: com.apple.launchservicesd; name: RoleUserInteractiveFocal; sourceEnvironment: 0x0>;
}

> Of course, if you have any other intuition on what could be special about your systems would help (e.g. special hardware like a keyboard, apps to remap keys, etc)

Before installing AltTab some months ago, I had changed System Preferences->Keyboard->Modifier Keys. However, these are now back to defaults ("Restore Defaults")

Also, I use an external keyboard: Corsair K63 wired. It is intended for Windows since it has a Windows key.

**Comment by lwouis** [comment]
> Indeed, it does seem related to the external keyboard because now that I'm testing with the built-in macOS keybaord, I do not have this problem.

Ok that seems to be the root cause right here. What key do you press on the keyboard? (photo of the keyboard would help visualize). I see in your preferences that you set the `⌥` key. This Windows keyboard may not map its `alt` key to `⌥` (`option` key). Can you use the key in other system shortcuts like if you right-click the Finder icon in the Dock, pressing `option` should change "Hide" to "Hide Others". You can also use karabiner companion app Karabiner-EventViewer. It's installed in the same folder as Karabiner-Elements when you [install Karabiner-Elements](https://karabiner-elements.pqrs.org/). Here is what pressing and releasing the `option` key is supposed to look like there:

<img width="986" alt="Screen Shot 2020-07-28 at 17 21 49" src="https://user-images.githubusercontent.com/106195/88638993-32713180-d0f7-11ea-9266-dc6591c4afc4.png">

**Comment by lwouis** [comment]
@ericjung any update?

**Comment by ericjung** [comment]
@lwouis thank you for following-up.

> What key do you press on the keyboard? (photo of the keyboard would help visualize)

I am pressing alt+tab. To remind you, that displays the list of application thumbnails and I can cycle through them with further alt+tab keypresses. However, releasing the keys does not activate/focus the highlighted thumbnail. It does nothing and I must click on the thumbnail to activate/focus it (and remove the thumnail images).

Here is a photo of the keyboard:
![IMG-4877](https://user-images.githubusercontent.com/5875610/89900057-7573e000-dba0-11ea-9fbf-995a3394d665.JPG)

> Can you use the key in other system shortcuts like if you right-click the Finder icon in the Dock, pressing option should change "Hide" to "Hide Others"

When I right-click on the Finder icon in the Dock and press `alt`, I do indeed see "Hide" change to "Hide Others"

> You can also use karabiner companion app Karabiner-EventViewer. It's installed in the same folder as Karabiner-Elements when you install Karabiner-Elements. 

I installed this, but it is asking to install a kernel extension in order to work properly. I'm a little scared of that. I'll do it if you still need more information. Please let me know.

Thank you!

**Comment by lwouis** [comment]
> I installed this, but it is asking to install a kernel extension in order to work properly. I'm a little scared of that. I'll do it if you still need more information. Please let me know.

Karabiner is a [successful and popular open-source mac app with 11k stars on Github](https://github.com/pqrs-org/Karabiner-Elements). I would advise you to install the kernel extension, but of course it's up to you in the end. Maybe before doing that, try this [online tool](https://w3c.github.io/uievents/tools/key-event-viewer.html). No install required for this one, just press and release the alt key and share what you get. It may be enough, or maybe not. This browser version can't access deep technical data like Karabiner-Viewer can, but it may be enough to diagnose why your alt key doesn't behave exactly like the internal keyboard alt-key.

Note that this is the swift code that returns `true` for your internal keyboard, but `false` for your external windows keyboard:

```swift
event.sr_keyEventType == .up && type == .flagsChanged && shortcut.keyCode == .none && event.modifierFlags.isDisjoint(with: shortcut.modifierFlags)
```

What I'm trying to diagnose if which of these conditions is different between both keyboards.

**Comment by ericjung** [comment]
OK, I have Karabiner working now. And I have some more important information. But first:

> Here is what pressing and releasing the option key is supposed to look like there:

This is what I get; same as you:
![alt-tab](https://user-images.githubusercontent.com/5875610/90044741-51d99400-dc8b-11ea-875d-557957ba999f.jpg)

**HOWEVER**, I only get that when I plug the external keyboard directly into the macbook. And alt-tab works correctly when I do that -- there is no bug.

I should have mentioned earlier, and I'm very sorry that I did not, that I am using [Synergy](https://symless.com/synergy) with this external keyboard connected to a linux Synergy host/server and the macbook running Synergy as a client.

So it looks like Synergy with alt-tab is the problem.

If I run Karabiner EventViewer with the external keyboard connected to the Synergy server and macbook running Synergy client, EventViewer is unable to capture keyboard events -- there is some conflict. So I'm not able to report what Synergy is sending to alt-tab with EventViewer.

Is there another way?

For reference, in case you happen to be familiar with Synergy, here is a simplified Synergy configuration I use (I removed 3 machines that are not macbooks; normally I have 6 machines sharing the keyboard/mouse through Synergy). This configuration includes 1 linux synergy server and 2 macbook synergy clients, both running alt-tab), and exhibits the behavior in this bug report.

```synergy configuration: grendel is linux mint and the server
bilestload.local is macOS and a client
n is macOS and a client
Keyboard is a Windows-style keyboard.
The important stuff is the shift/ctrl/alt/meta/super mappings

section: screens
  bilestoad.local:
    # macbook running synergy client
    halfDuplexCapsLock = false
    halfDuplexNumLock = false
    halfDuplexScrollLock = false
    xtestIsXineramaUnaware = false
    switchCorners = none
    switchCornerSize = 0
    shift = shift
    ctrl = meta
    alt = alt
    meta = super
    super = alt
  grendel:
   # linux running synergy server with external keyboard connected
    halfDuplexCapsLock = false
    halfDuplexNumLock = false
    halfDuplexScrollLock = false
    xtestIsXineramaUnaware = false
    switchCorners = none
    switchCornerSize = 0
  n:
    # macbook running synergy client
    halfDuplexCapsLock = false
    halfDuplexNumLock = false
    halfDuplexScrollLock = false
    xtestIsXineramaUnaware = false
    switchCorners = none
    switchCornerSize = 0
    shift = shift
    ctrl = meta
    alt = alt
    meta = super
    super = alt
end

section: aliases
end

section: links
  bilestoad.local:
    right = n
  grendel:
    left = n
  n:
  right = grendel
  left = bilestoad.local
end


section: options
  relativeMouseMoves = false
  screenSaverSync = true
  win32KeepForeground = false
  clipboardSharing = true
  switchCorners = none
  switchCornerSize = 0
end```

**Comment by lwouis** [comment]
Checklist:

* External keyboard
* Windows keyboard layout
* Synergy virtualizing keyboard inputs to 2 different macbooks

This setup is nightmare fuel for debugging what's happening. Are you trying to kill me? 🙉

I'm joking lol but damn am I discovering exotic setups through this project. You should change the Input Source to Arabic now, to make things a bit more interesting ;p

> If I run Karabiner EventViewer with the external keyboard connected to the Synergy server and macbook running Synergy client, EventViewer is unable to capture keyboard events -- there is some conflict. So I'm not able to report what Synergy is sending to alt-tab with EventViewer.

What's happening exactly? Are presses of `option` simply not showing in the Viewer? That's not normal. Can you try on the [online tool](https://w3c.github.io/uievents/tools/key-event-viewer.html) I mentioned previously? Does the tool output this, with and without synergy?

![image](https://user-images.githubusercontent.com/106195/90054237-89d9e900-dd16-11ea-8217-10ca97de2530.png)

Regarding your synergy conf, I see these remaps:

```
shift = shift
ctrl = meta
alt = alt
meta = super
super = alt
```

What if you remove these and keep the keys one for one instead, and test? The above config is tricky as the order of execution matters. For example `ctrl = meta` then `meta = super` which implies `ctrl = super`. Not sure what you're doing here but that's not trivial.

**Comment by ericjung** [comment]
> This setup is nightmare fuel for debugging what's happening. Are you trying to kill me? 

Sorry.

> Synergy virtualizing keyboard inputs to 2 different macbooks

I don't think this is relevant: same problem with just 1 macbook as a synergy client instead of 2

> You should change the Input Source to Arabic now

RTL languages are a bitch to support and still make a product look good. I know from personal experience.

> What's happening exactly? Are presses of option simply not showing in the Viewer?

Correct. However, if it quit Alt-Tab, then I 

> Can you try on the online tool I mentioned previously? Does the tool output this, with and without synergy?

Here are my results with [Keyboard Event Viewer](https://w3c.github.io/uievents/tools/key-event-viewer.html) including the built-in macbook keyboard, external Corsair K63 wired keyboard connected to linux box running synergy server, and an [Amazon Basics ku-0833](https://www.amazon.com/AmazonBasics-Wired-Computer-Keyboard-Bundle/dp/B00B7GV802) external wired keyboard connected directly to the macbook pro (well, through a usb hub connected directly to the macbook pro USB-C port):

> What if you remove these and keep the keys one for one instead, and test?

OK, synergy.conf has all key mappings removed that you mentioned and synergy has been restarted on both devices.

| key pressed | built-in macbook pro keyboard  | USB-connected [Amazon Basics ku-0833](https://www.amazon.com/AmazonBasics-Wired-Computer-Keyboard-Bundle/dp/B00B7GV802) | [Corsair K63 ](https://www.corsair.com/us/en/Categories/Products/Gaming-Keyboards/Mechanical-Gaming-Keyboards/K63-Compact-Mechanical-Gaming-Keyboard-%E2%80%94-CHERRY%C2%AE-MX-Red/p/CH-9115020-NA)synergy-connected keyboard |
| ------------- | -------------| ------------- | ------------- |
| left ctrl  | 17 ControlLeft |  17 ControlLeft  | nothing |
| left shift  | 16 ShiftLeft |  16 ShiftLeft  | nothing |
| left option  | 18 AltLeft |  key does not exist  | key does not exist |
| left command  | 81 MetaLeft |  key does not exist  | key does not exist |
| windows key  | key does not exist | 91 MetaLeft  | nothing |
| left alt  | key does not exist | 18 AltLeft  | nothing |

Swapping the two external keyboards keyboard USB connected vs Synergy-connected makes no difference in output.

Also of interest:

**With Alt-Tab not running**: ctrl+tab on the usb-connected keyboard switches applications just like command+tab does on the built-in keyboard. But ctrl+tab on the synergy-connected keyboard does nothing.

**With Alt-Tab running**: alt+tab on the usb-connected keyboard switches applications as you'd expect, and releasing alt-tab activates/focused the highlighted application thumbnail. alt+tab on the synergy-connected keyboard exhibits the behavior I first described in this bug; i.e., changing synergy.conf did not affect this behavior.

**Comment by lwouis** [comment]
> With Alt-Tab not running: ctrl+tab on the usb-connected keyboard switches applications just like command+tab does on the built-in keyboard. But ctrl+tab on the synergy-connected keyboard does nothing.

It seems clear that this is a result of Synergy. Could you please try contacting their support and asking them if they know why you can't do ctrl+tab, or why the option and command keys are not recognized on the web? I would suggest linking this thread in the support message, especially the latest messages.

Also, could you please try [the build I linked on this other ticket](https://github.com/lwouis/alt-tab-macos/issues/157#issuecomment-673037037)? It's a preview of a complete rework of the keyboard support in AltTab, to avoid issues with macOS Secure Input. I'm curious to known if that build reacts differently, or if the events are simply never sent by synergy.

**Comment by ericjung** [comment]
> It seems clear that this is a result of Synergy.

Sorry for wasting yout time. I did not know it was a Synergy problem before we entered this rabbit hole. I apologize.

> Also, could you please try the build I linked on this other ticket?

Yes

**Comment by lwouis** [comment]
Closing this ticket as it is an issue with Synergy.app. @ericjung please let us know if they reply and say it's on AltTab's side somehow, then we will investigate again with their feedback.

Also, did the latest version change any of that? The keyboard input code was completely replaced in v5. It may or may not fix your issue

**Comment by ericjung** [comment]
I have some amazing updates.

**First:**

> With Alt-Tab not running: ctrl+tab on the usb-connected keyboard switches applications just like command+tab does on the built-in keyboard. But ctrl+tab on the synergy-connected keyboard does nothing.

This is no longer the case. After restarting all machines and ignoring the issue for 12 days now, when I tried stop Alt-Tab and AND use a synergy-connected keyboard to the macbook, ctrl+tab switches applications just like command+tab on the built-in keyboard. Synergy versions did not change but it is possible Alt-Tab updated.

> It seems clear that this is a result of Synergy. Could you please try contacting their support and asking them if they know why you can't do ctrl+tab, or why the option and command keys are not recognized on the web?

I planned to open this ticket today at https://github.com/symless/synergy-core but after re-testing it seems not a Synergy problem.

**Second:**

> Also, could you please try the build I linked on this other ticket? 

[This build](https://github.com/lwouis/alt-tab-macos/files/5070824/AltTab.app.zip) completely fixes the problem I originally mentioned in this ticket.

**Comment by ericjung** [comment]
> This build completely fixes the problem I originally mentioned in this ticket.

OK, this is really strange. As I mentioned before, I have 2 macbooks. The problem is completely fixed with this build on *only one* of the macbooks. Not both. They are  both using the same Synergy configurations. They are both using macOS Catalina 10.15.6. I'll try restarting the faulty one.

**Comment by lwouis** [comment]
Could you use the public release instead? (by clicking the [download button at the top of the website](https://alt-tab-macos.netlify.app/)). That way we are on stable grounds

**Comment by ericjung** [comment]
The problem is fixed on both machines using AltTab 5.3.0.
THANK YOU!

**Comment by lwouis** [comment]
Very happy to hear that @ericjung! 💯
